core/src/modules/configWriter.ts

```diff
@@ -18,7 +21,32 @@ type WritableConfigObject = {
   object: ConfigurationObject;
 };
 
-let FILES_TO_DELETE: string[] = [];
+type CachedConfigFile = {
+  absFilePath: string;
+  objects: ConfigurationObject[];
+};
+
+/**
+ * Loading
+ * ----------------------------------------
+ * 1. Loader tells Writer to cache all files it read.
+ * 2. Writer caches files that are writable (i.e. not locked). It ignores locked
+ *    files.
+ * 3. Individual loaders ask the writer for a lock key. Writer responds with the
+ *    key unless it has a cached object.
+ *
+ *
+ * Writing
+ * ----------------------------------------
+ * 1. Delete all files in the cache.
+ * 2. Query the database for all writable objects. These are UNLOCKED Apps,
+ *    Sources (and their Schedules), Properties, Groups, and Destinations.
+ * 3. Each config object is retrieved from the model, written to file, and then
+ *    cached.
+ *
+ */
```

Comment:
This felt like a lot to have in this file. It really helped me work through it and pull it together. I didn't come up with a better place to put this, but happy to move or remove it.